lib/bmg/operator/minus.rb

```diff
@@ -0,0 +1,43 @@
+module Bmg
+  module Operator
+    #
+    # Minus operator.
+    #
+    # Returns all tuples which are in the left operand but not 
+    # in the right operand.
+    #
+    # This implementation is actually a NAry-Minus, since it handles
+    # an arbitrary number of operands.
+    #
+    class Minus
+      include Operator::Nary
+
+      def initialize(type, operands)
+        @type = type
+        @operands = operands
+      end
+
+    public
+
+      def all?
+        false 
+      end
+
+      def each(&bl)
+        return to_enum unless block_given?
+        initial = operands[0].to_a
+        tuples = operands.drop(1).inject(initial) do |agg, op|
+          agg - op.to_a
+        end
```

Comment:
Definitely not the optimal implementation, but I suppose we don't really care about it (?)

Reply:
Maybe using Ruby's `Set`? Which I presume uses object hashes? That should be reasonably good.
I'll do a perf test to check that it's actually faster (although the code will be very similar).

Reply:
Fixed this. My benchmark went from ~25s to ~1s.

See: https://gist.github.com/felixyz/7151d6acc1ef3d0d674caeacc9538862

Slight worry: the reported remaining number of tuples went down from ~55k to ~42k.

The optimization commit is here: https://github.com/enspirit/bmg/pull/45/commits/d75847b4569e4207e581861c9cb88f14e9b0f308